src/Validation/Validation.php

```diff
@@ -733,26 +730,38 @@ public static function decimal(mixed $check, int|bool|null $places = null, ?stri
         }
 
         if ($regex === null) {
-            $lnum = '[0-9]+';
-            $dnum = "[0-9]*[\.]{$lnum}";
-            $sign = '[+-]?';
-            $exp = "(?:[eE]{$sign}{$lnum})?";
-
-            if ($places === null) {
-                $regex = "/^{$sign}(?:{$lnum}|{$dnum}){$exp}$/";
-            } elseif ($places === true) {
-                if (is_float($check) && floor($check) === $check) {
-                    $check = sprintf('%.1f', $check);
-                }
-                $regex = "/^{$sign}{$dnum}{$exp}$/";
-            } else {
-                $places = '[0-9]{' . $places . '}';
-                $dnum = "(?:[0-9]*[\.]{$places}|{$lnum}[\.]{$places})";
-                $regex = "/^{$sign}{$dnum}{$exp}$/";
-            }
+            [$check, $regex] = static::getDecimalRegex($check, $places);
+        }
+
+        return static::_check($check, $regex);
+    }
+
+    /**
+     * Checks that a value is a valid decimal. Both the sign and exponent are optional.
+     *
+     * The currently set locale is used to determine the decimal and thousands separator of the given number.
+     *
+     * Valid Places:
+     *
+     * - null => Any number of decimal places, including none. The '.' is not required.
```

Comment:
Think this should be precision not places.

Reply:
I have used the same argument name as the existing method but I agree, `precision` would be a more accurate name.